### PR TITLE
feat(ui): Collapsable sidebar with filters

### PR DESCRIPTION
**Pull request by rbreeze**
CleanShot 2022-09-16 at 14 09 12@2x
CleanShot 2022-09-16 at 14 09 21@2x

CleanShot 2022-09-16 at 14 09 02@2x
CleanShot 2022-09-16 at 14 09 17@2x

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [ ] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

### CODE REVIEW

**Review by jessesuen — COMMENTED** [review]
Why is there an outline surrounding these filters? Can they be removed?

image

**Comment by rbreeze** [comment]
@jessesuen  How does this look? 

CleanShot 2022-09-16 at 17 14 39@2x
CleanShot 2022-09-16 at 17 14 54@2x

File: ui/src/app/shared/components/layout/layout.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import * as React from 'react';
+import {Sidebar} from '../../../sidebar/sidebar';
+
+require('./layout.scss');
+
+export interface LayoutProps {
```
Comment (ashutosh16) [inline ui/src/app/shared/components/layout/layout.tsx:6]: Is the layout component the main layout or sidebar layout?

Comment (rbreeze) [reply ui/src/app/shared/components/layout/layout.tsx:6]: Layout is a main layout, copied from `argo-ui`

**Review by alexmt — APPROVED** [review]
LGTM!

**Pull request merged**